app/javascript/pages/Admin/Purchases/Show.tsx

```diff
@@ -0,0 +1,27 @@
+import { usePage } from "@inertiajs/react";
+import React from "react";
+import { cast } from "ts-safe-cast";
+
+import Product, { type Product as ProductType } from "$app/components/Admin/Products/Product";
+import Purchase, { type Purchase as PurchaseType } from "$app/components/Admin/Purchase";
+import User, { type User as UserType } from "$app/components/Admin/Users/User";
+
+type AdminPurchaseProps = {
+  user: UserType;
+  product: ProductType;
+  purchase: PurchaseType;
+};
+
+const AdminPurchasesShow = () => {
+  const { user, product, purchase } = cast<AdminPurchaseProps>(usePage().props);
+
+  return (
+    <div className="paragraphs">
+      <Purchase purchase={purchase} />
```

Comment:
This is why I don't like default exports (I complained about it to Emmanuel in another PR 😆) - I was just looking at a component called `AdminPurchase`, and now I see a `Purchase` component used here and think it's a different wrapper or something

(I don't really feel that strongly, but I'd argue `import { AdminProduct, type Product } from ".../Admin/Product"` is cleaner than the alias here)